src/import/crd.ts

```diff
@@ -102,49 +105,29 @@ export class CustomResourceDefinition {
 }
 
 export class ImportCustomResourceDefinition extends ImportBase {
-  public static async match(importSpec: ImportSpec): Promise<undefined | ManifestObjectDefinition[]> {
+  public static async fromSpec(importSpec: ImportSpec): Promise<ImportCustomResourceDefinition> {
     const { source } = importSpec;
     const manifest = await download(source);
-    return yaml.parseAllDocuments(manifest).map((doc: yaml.Document) => doc.toJSON());
+    return new ImportCustomResourceDefinition(safeParseCrds(manifest));
   }
 
   private readonly groups: Record<string, CustomResourceDefinition[]> = { };
 
-  constructor(manifest: ManifestObjectDefinition[]) {
+  private constructor(manifest: ManifestObjectDefinition[]) {
     super();
 
     const crds: Record<string, CustomResourceDefinition> = { };
     const groups: Record<string, CustomResourceDefinition[]> = { };
 
-    const extractCRDs = (objects: ManifestObjectDefinition[] = []) => {
```

Comment:
This was moved to the yaml parsing logic since we need to perform schema validation on each CRD.